### PR TITLE
Add rdflib-jsonld module?

### PR DESCRIPTION
**Pull request by matentzn**

### CODE REVIEW

**Comment by cthoyt** [comment]
Seems like this didn't do the trick

**Comment by cthoyt** [comment]
Weird, the use of https://pypi.org/project/rdflib-jsonld/0.5.0/ didn't work. Maybe it has to be marked as <= instead of ==. I'm testing locally now

**Comment by cthoyt** [comment]
It might be okay to not restrict rdflib-jsonld, it seems to work with rdflib==5.0.0 when it installs rdflib-jsonld==0.6.2 (contrary to what their readme says, which seems to be specifically about the old version python3.6)

**Comment by matentzn** [comment]
Super odd.. Locally the tests all work for me, using the exact versions indicated here.

**Comment by cthoyt** [comment]
I'm experiencing mostly the same thing, except `rdflib-pyld-compat` isn't necessary to get it to work

**Comment by matentzn** [comment]
could it be that setup.cfg somehow works offs a different version of pip, one that doesn't know old versions?

**Comment by cthoyt** [comment]
Setuptools >= 58 (released september 5th/three weeks ago) breaks usage of 2to3, so I think we have to figure out how to downgrade setuptools

Ref: https://exerror.com/error-in-mongoengine-setup-command-use_2to3-is-invalid/

File: setup.cfg [hunk 1]
```diff
@@ -64,7 +64,7 @@ install_requires =
     recommonmark>=0.7
     scikit_learn
     scipy
-    setuptools
+    setuptools<=57.5.0
```
Comment (cthoyt) [inline setup.cfg:67]: oh weird, typically setuptools isn't actually a requirement for using the code. I'm surprised this is listed here

**Comment by matentzn** [comment]
The primary problem is that some of our core dependencies use rdflib 5. This is what I believe causes the whole problem now - rdflib-jsonld has merged into rdflib 6 and is now obsolete.

**Comment by cthoyt** [comment]
Oh, yes. This is the case. rdflib-jsonld has been "tombstoned" as of version 0.6.0 which means you can download it but now it doesn't do anything. Should we just put more effort into upgrading to rdflib 6 instead?

**Comment by matentzn** [comment]
@hrshdhgd can you look into this? 

The last thing we want to do, but we will if we must, is go try migrating linkml to rdflib 6...

I pinged them a bit on that front: https://github.com/linkml/linkml/issues/291

 will also talk to @cmungall later today about this issue.

**Comment by matentzn** [comment]
I also left a comment at rdflib, just in case: https://github.com/RDFLib/rdflib/issues/1417

**Comment by hrshdhgd** [comment]
```
ERROR: Could not find a version that satisfies the requirement rdflib-jsonld==0.5.0 (from sssom[test]) (from versions: 0.2, 0.3, 0.4.0, 0.5.0, 0.6.0, 0.6.1, 0.6.2)
59
ERROR: No matching distribution found for rdflib-jsonld==0.5.0
```

So weird .... the version is listed right there

**Comment by cthoyt** [comment]
@hrshdhgd scoot up a bit - the issue is that it can't install rdflib-jsonld==0.5.0 properly, not that it can't find it. it's the 2to3 issue

**Comment by hrshdhgd** [comment]
0.6.0 did the trick!

File: setup.cfg [hunk 2]
```diff
@@ -57,11 +57,12 @@ install_requires =
...
     pandasql
     pyyaml
-    rdflib
+    rdflib==5.0.0
+    rdflib-jsonld==0.6.0
     recommonmark>=0.7
     scikit_learn
     scipy
-    setuptools
+    setuptools<58.0
```
Comment (matentzn) [inline setup.cfg:65]: is this still necessary?

File: tox.ini [hunk 3]
```diff
@@ -8,6 +8,8 @@ envlist =
     lint
     flake8
     py
+requires =
+   setuptools<58.0
```
Comment (matentzn) [inline tox.ini:12]: make sure this is still necessary/

File: tox.ini [hunk 4]
```diff
@@ -16,6 +18,7 @@ extras =
     test
 description = Run unit tests with pytest. This is a special environment that does not get a name, and
               can be referenced with "py".
+skipsdist = False
```
Comment (matentzn) [inline tox.ini:21]: Is this necessary, who put it there?

**Comment by matentzn** [comment]
Thank you both :) Awesome. Made a few comments!

File: tox.ini (hunk 4 above)
Comment (hrshdhgd) [reply tox.ini:21]: I put it there. Was testing it..not required. Taking it off.

File: tox.ini (hunk 3 above)
Comment (hrshdhgd) [reply tox.ini]: nope, taking it out!

File: setup.cfg (hunk 2 above)
Comment (hrshdhgd) [reply setup.cfg:65]: Yes.

Comment (matentzn) [reply setup.cfg:65]: Why? Just so we have it documented :) thank you!

**Comment by cmungall** [comment]
Is this tombstoning standard practice?
https://github.com/RDFLib/rdflib-jsonld/releases/tag/0.6.2

So they scooped out all functionality and made a new.. minor release? To force people to switch?

**Comment by matentzn** [comment]
@hrshdhgd why did you remove the version on rdf lib jsonld? This should fail. The question is why we need the version on setuptools

**Comment by hrshdhgd** [comment]
> @hrshdhgd why did you remove the version on rdf lib jsonld? This should fail. The question is why we need the version on setuptools

Just to document the repercussions:

On removing version for redflib-jsonld, we get the error:
```
No module named 'rdflib_jsonld.parser'
```

On removing the versioning pin for setuptools, it all works.

File: setup.cfg (hunk 2 above)
Comment (hrshdhgd) [reply setup.cfg]: My bad, it is not needed. Took it off.

**Comment by matentzn** [comment]
Perfect thanks :)

**Comment by matentzn** [comment]
You can now merge this into the other open PRs to get them to pass again..

**Comment by matentzn** [comment]
Nice one man.

**Comment by cmungall** [comment]
also excluded tombstoned on linkml: https://github.com/linkml/linkml/pull/352

**Pull request merged**